### DISCUSSION TITLE
Looking for a proper `dap.configurations.cpp` config for inspecting coredump (C++).

### DISCUSSION THREAD

**Discussion by Hastyshell** [mfussenegger/nvim-dap#1224] [Q&A]
Currently, my configuration as below works well with launching an executable and attaching to a process. However, I was confused when I tried to add a new configuration for inspecting a coredump file.
```
dap.configurations.cpp = {
        {
          name = "Launch executable",
          type = "codelldb",
          request = "launch",
          program = function()
            return vim.fn.input("Path to executable: ", vim.fn.getcwd() .. "/", "file")
          end,
          cwd = "${workspaceFolder}",
          stopOnEntry = false,
        },
        {
          name = "Attach to process",
          type = "codelldb",
          request = "attach",
          processId = require("dap.utils").pick_process,
          cwd = "${workspaceFolder}",
        },
}
```
Are there any ways to `dap.configurations.cpp` configuration like below equally? `requst = "custom"` seems not supported.
```
{
            "type": "lldb",
            "request": "custom",
            "name": "CodeLLDB core",
            "initCommands": [
                "target create ${workspaceFolder}/demo -c ${input:coreFileName}"
            ]
 }
```

**Comment by mfussenegger** [comment]
nvim-dap doesn't support a `custom` `request`. This sounds like a feature of the vscode extension?

**Reply by Hastyshell** [reply]
@mfussenegger Thanks for ur reply! Yes, actually it is the codelldb config option for user to use lldb command directly to create a debug session. Does nvim-dap support a similar way to create a debug session with customized lldb command (like "lldb target create -c ${corefile}") other than `launch` or `attach`? Since coredump file seems not debuggy directly with either `launch` or `attach` operation.

**Reply by mfussenegger** [reply]
I'd need some explanation what `custom` is supposed to do.

nvim-dap supports starting processes, or connecting via tcp/pipe.

**Reply by Hastyshell** [reply]
@mfussenegger `custom` seems to be a way for starting a debug session neither by starting a process with an executable nor connecting via tcp/pipe. It allows users to start the debug session with specified commands of debuger on specified targets.

For the example of a coredump file, which is not a normal executable file but a file containing the memory states of the crashed program, we could not simply start a debug session just in `launch` or `attach` way with `nvim-dap`, but from the perspective of debug adaptors, like `codelldb`, it supports an particular way allowing users to fully specify how the debug session is initiated with debuger's commands. [see](https://github.com/vadimcn/codelldb/blob/master/MANUAL.md#custom-launch). [and](https://github.com/vadimcn/codelldb/blob/master/MANUAL.md#inspecting-a-core-dump).

So is it necessary to introduce such an interface to nvim-dap for higher flexibility? Or there might be some ways we could have already been able to deal with those cases, just with `launch` and `attach`? Thanks a lot!
